cpp/src/IceStorm/Service.cpp

```diff
@@ -248,7 +248,7 @@ ServiceI::start(const string& serviceName, const CommunicatorPtr& communicator,
                     throw IceBox::FailureException(__FILE__, __LINE__, "IceGrid deployment is incorrect");
                 }
 
-                int nodeid = atoi(adapterId.substr(start, end - start).c_str());
+                int nodeid = static_cast<int>(strtol(adapterId.substr(start, end - start).c_str(), nullptr, 10));
```

Comment:
Shouldn't this use `std::stoi` instead of doing a static_cast to int

https://en.cppreference.com/w/cpp/string/basic_string/stol

Reply:
Yes, you're right, fixed.